File: plasmid/source/core-plugins/eln-lims/1/as/webapps/eln-lims/html/lib/openbis/js/v3/as/dto/global/fetchoptions/GlobalSearchObjectFetchOptions.js
define([ "require", "stjs", "as/dto/common/fetchoptions/FetchOptions", "as/dto/experiment/fetchoptions/ExperimentFetchOptions", "as/dto/sample/fetchoptions/SampleFetchOptions",
		"as/dto/dataset/fetchoptions/DataSetFetchOptions", "as/dto/material/fetchoptions/MaterialFetchOptions", "as/dto/global/fetchoptions/GlobalSearchObjectSortOptions" ], function(require, stjs,
		FetchOptions) {
	var GlobalSearchObjectFetchOptions = function() {
	};
	stjs.extend(GlobalSearchObjectFetchOptions, FetchOptions, [ FetchOptions ], function(constructor, prototype) {
		prototype['@type'] = 'as.dto.global.fetchoptions.GlobalSearchObjectFetchOptions';
		constructor.serialVersionUID = 1;
		prototype.experiment = null;
		prototype.sample = null;
		prototype.dataSet = null;
		prototype.material = null;
		prototype.sort = null;

		prototype.withExperiment = function() {
			if (this.experiment == null) {
				var ExperimentFetchOptions = require("as/dto/experiment/fetchoptions/ExperimentFetchOptions");
				this.experiment = new ExperimentFetchOptions();
			}
			return this.experiment;
		};
		prototype.withExperimentUsing = function(fetchOptions) {
			return this.experiment = fetchOptions;
		};
		prototype.hasExperiment = function() {
			return this.experiment != null;
		};

		prototype.withSample = function() {
			if (this.sample == null) {
				var SampleFetchOptions = require("as/dto/sample/fetchoptions/SampleFetchOptions");
				this.sample = new SampleFetchOptions();
			}
			return this.sample;
		};
		prototype.withSampleUsing = function(fetchOptions) {
			return this.sample = fetchOptions;
		};
		prototype.hasSample = function() {
			return this.sample != null;
		};

		prototype.withDataSet = function() {
			if (this.dataSet == null) {
				var DataSetFetchOptions = require("as/dto/dataset/fetchoptions/DataSetFetchOptions");
				this.dataSet = new DataSetFetchOptions();
			}
			return this.dataSet;
		};
		prototype.withDataSetUsing = function(fetchOptions) {
			return this.dataSet = fetchOptions;
		};
		prototype.hasDataSet = function() {
			return this.dataSet != null;
		};

		prototype.withMaterial = function() {
			if (this.material == null) {
				var MaterialFetchOptions = require("as/dto/material/fetchoptions/MaterialFetchOptions");
				this.material = new MaterialFetchOptions();
			}
			return this.material;
		};
		prototype.withMaterialUsing = function(fetchOptions) {
			return this.material = fetchOptions;
		};
		prototype.hasMaterial = function() {
			return this.material != null;
		};

		prototype.sortBy = function() {
			if (this.sort == null) {
				var GlobalSearchObjectSortOptions = require("as/dto/global/fetchoptions/GlobalSearchObjectSortOptions");
				this.sort = new GlobalSearchObjectSortOptions();
			}
			return this.sort;
		};
		prototype.getSortBy = function() {
			return this.sort;
		};
	}, {
		experiment : "ExperimentFetchOptions",
		sample : "SampleFetchOptions",
		dataSet : "DataSetFetchOptions",
		material : "MaterialFetchOptions",
		sort : "GlobalSearchObjectSortOptions"
	});
	return GlobalSearchObjectFetchOptions;
})

File: openbis_api/source/java/ch/ethz/sis/openbis/generic/asapi/v3/dto/global/fetchoptions/GlobalSearchObjectFetchOptions.java
/*
 * Copyright 2014 ETH Zuerich, CISD
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package ch.ethz.sis.openbis.generic.asapi.v3.dto.global.fetchoptions;

import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.fetchoptions.FetchOptions;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.fetchoptions.FetchOptionsToStringBuilder;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.dataset.fetchoptions.DataSetFetchOptions;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.experiment.fetchoptions.ExperimentFetchOptions;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.global.GlobalSearchObject;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.material.fetchoptions.MaterialFetchOptions;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.sample.fetchoptions.SampleFetchOptions;
import ch.systemsx.cisd.base.annotation.JsonObject;
import com.fasterxml.jackson.annotation.JsonProperty;
import java.io.Serializable;

/*
 * Class automatically generated with DtoGenerator
 */
@JsonObject("as.dto.global.fetchoptions.GlobalSearchObjectFetchOptions")
public class GlobalSearchObjectFetchOptions extends FetchOptions<GlobalSearchObject> implements Serializable
{
    private static final long serialVersionUID = 1L;

    @JsonProperty
    private ExperimentFetchOptions experiment;

    @JsonProperty
    private SampleFetchOptions sample;

    @JsonProperty
    private DataSetFetchOptions dataSet;

    @JsonProperty
    private MaterialFetchOptions material;

    @JsonProperty
    private GlobalSearchObjectSortOptions sort;

    // Method automatically generated with DtoGenerator
    public ExperimentFetchOptions withExperiment()
    {
        if (experiment == null)
        {
            experiment = new ExperimentFetchOptions();
        }
        return experiment;
    }

    // Method automatically generated with DtoGenerator
    public ExperimentFetchOptions withExperimentUsing(ExperimentFetchOptions fetchOptions)
    {
        return experiment = fetchOptions;
    }

    // Method automatically generated with DtoGenerator
    public boolean hasExperiment()
    {
        return experiment != null;
    }

    // Method automatically generated with DtoGenerator
    public SampleFetchOptions withSample()
    {
        if (sample == null)
        {
            sample = new SampleFetchOptions();
        }
        return sample;
    }

    // Method automatically generated with DtoGenerator
    public SampleFetchOptions withSampleUsing(SampleFetchOptions fetchOptions)
    {
        return sample = fetchOptions;
    }

    // Method automatically generated with DtoGenerator
    public boolean hasSample()
    {
        return sample != null;
    }

    // Method automatically generated with DtoGenerator
    public DataSetFetchOptions withDataSet()
    {
        if (dataSet == null)
        {
            dataSet = new DataSetFetchOptions();
        }
        return dataSet;
    }

    // Method automatically generated with DtoGenerator
    public DataSetFetchOptions withDataSetUsing(DataSetFetchOptions fetchOptions)
    {
        return dataSet = fetchOptions;
    }

    // Method automatically generated with DtoGenerator
    public boolean hasDataSet()
    {
        return dataSet != null;
    }

    // Method automatically generated with DtoGenerator
    public MaterialFetchOptions withMaterial()
    {
        if (material == null)
        {
            material = new MaterialFetchOptions();
        }
        return material;
    }

    // Method automatically generated with DtoGenerator
    public MaterialFetchOptions withMaterialUsing(MaterialFetchOptions fetchOptions)
    {
        return material = fetchOptions;
    }

    // Method automatically generated with DtoGenerator
    public boolean hasMaterial()
    {
        return material != null;
    }

    // Method automatically generated with DtoGenerator
    @Override
    public GlobalSearchObjectSortOptions sortBy()
    {
        if (sort == null)
        {
            sort = new GlobalSearchObjectSortOptions();
        }
        return sort;
    }

    // Method automatically generated with DtoGenerator
    @Override
    public GlobalSearchObjectSortOptions getSortBy()
    {
        return sort;
    }
    @Override
    protected FetchOptionsToStringBuilder getFetchOptionsStringBuilder()
    {
        FetchOptionsToStringBuilder f = new FetchOptionsToStringBuilder("GlobalSearchObject", this);
        f.addFetchOption("Experiment", experiment);
        f.addFetchOption("Sample", sample);
        f.addFetchOption("DataSet", dataSet);
        f.addFetchOption("Material", material);
        return f;
    }

}
